pkg/controller/workload/mpijob/mpijob_webhook.go

```diff
@@ -112,7 +113,9 @@ func (w *MPIJobWebhook) ValidateUpdate(ctx context.Context, oldObj, newObj runti
 func validateUpdate(oldJob, newJob *kubeflow.MPIJob) field.ErrorList {
 	allErrs := validateCreate(newJob)
 
-	if !*newJob.Spec.RunPolicy.Suspend && (queueName(oldJob) != queueName(newJob)) {
+	oldMPIJob := MPIJob{*oldJob}
+	newMPIJob := MPIJob{*newJob}
```

Comment:
I still have the feeling that using an alias would save this kind of lines.

Reply:
I agree that methods like QueueName, or ParentWorkloadName belong to the Job indeed. However, methods for reconciling seem outside. Maybe we should have two types and split the methods between them. For example GenericJob and GenericJobReconciler. Then, I would be happy to use type aliases. I think this is something I would like to work as the next PR.

Reply:
sg